client/my-sites/stats/activity-log-tasklist/plugins-to-update.jsx

```diff
@@ -0,0 +1,56 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { unionBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getPluginsWithUpdates } from 'state/plugins/installed/selectors';
+
+export default WrappedComponent => {
+	class PluginsToUpdate extends Component {
+		static propTypes = {
+			siteId: PropTypes.number,
+			pluginsWithUpdates: PropTypes.arrayOf( PropTypes.object ),
+		};
+
+		state = {
+			// Plugins with pending updates
+			pluginsWithUpdates: [],
+			// Plugins already updated + those with pending updates
+			allPluginsToUpdate: [],
+		};
+
+		componentWillReceiveProps( nextProps ) {
+			if ( this.state.pluginsWithUpdates.length < nextProps.pluginsWithUpdates.length ) {
+				this.setState( {
+					pluginsWithUpdates: nextProps.pluginsWithUpdates,
+					allPluginsToUpdate: unionBy(
+						nextProps.pluginsWithUpdates,
+						this.state.allPluginsToUpdate,
+						'slug'
+					),
+				} );
+			}
+		}
+
+		render() {
+			return (
+				<WrappedComponent
```

Comment:
Note here that we're wiping out any props passed to the wrapper and only sending these three. We may want to send `this.props` as well, or those which this wrapper doesn't intercept

```js
const { pluginsWithUpdates, siteId, ...props } = this.props;

return <WrappedComponent { ...props } siteId={ siteId } … />
```

Reply:
I intentionally wanted to pass only these since I don't need anything else.

Reply:
It's probably worth a big comment expressing that then or worth reconsidering because I don't think that's what most people will expect when they wrap a component.

I know that the design of this grew out of a single component into a split. If we don't want the wrapper to be used elsewhere we can also consider defining it in the same file as the wrapped component. That still has merit and removes some of the burden of not coupling the two.

Reply:
Passed all props as `{ ...this.props }` to comply with expectations.